src/benchmarks/micro/runtime/StoreBlock/StoreBlock.tt

```diff
@@ -0,0 +1,114 @@
+﻿<#@ template debug="false" hostspecific="true" language="C#" #>
+<#@ output extension=".cs" #>
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+[BenchmarkCategory(Categories.Runtime, Categories.JIT)]
+public class StoreBlock
+{
+    const int OperationsPerInvoke = 1000;
+
+    byte[] _srcData = new byte[1024];
+    byte[] _dstData = new byte[1024];
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        for (int i = 0; i < _srcData.Length; i++)
+            _srcData[i] = (byte)(i % 256);
+
+        for (int i = 0; i < _dstData.Length; i++)
+            _dstData[i] = (byte)(i % 256);
+    }
+<#
+    for (int byteCount = 8; byteCount <= 128; byteCount += 8)
+    {
+#>
+
+    [StructLayout(LayoutKind.Explicit, Size=<#= byteCount #>)]
+    struct Struct<#= byteCount #>
+    {
+    }
+
+    Struct<#= byteCount #> fld<#= byteCount #>;
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public void InitBlockAllZerosHeap<#= byteCount #>()
+    {
+        for (int i = 0; i < OperationsPerInvoke; i++)
+        {
+            Unsafe.InitBlock(ref _dstData[0], 0, <#= byteCount #>);
+        }
+    }
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public unsafe void InitBlockAllZerosLocalAddr<#= byteCount #>()
+    {
+        Struct<#= byteCount #> dstLcl;
+
+        for (int i = 0; i < OperationsPerInvoke; i++)
+        {
+            Unsafe.InitBlock(&dstLcl, 0, <#= byteCount #>);
+        }
+
+        fld<#= byteCount #> = dstLcl;
+    }
```

Comment:
Are we executing a different `Unsafe.InitBlock` code path compared to the `InitBlockAllZerosHeap` above?

https://github.com/dotnet/performance/blob/main/docs/microbenchmark-design-guidelines.md#test-cases

If not, we should have a single `Unsafe.InitBlock`  benchmark for given `byteCount`

```suggestion
```

Reply:
> Are we executing a different Unsafe.InitBlock code path compared to the InitBlockAllZerosHeap above?

At the moment they are the same.

However, I have been working on a JIT optimization on Arm64 to make a use of SIMD instructions for `InitBlock`/`CopyBlock`- see https://github.com/dotnet/runtime/pull/61030.

If/when that PR is merged - the JIT can utilize 16-byte alignment of `sp` and knowledge about distance between `fp` and `sp` and use 16-byte store SIMD operations for `InitBlock*LocalAddr`/`CopyBlock*LocalAddr`. In that case, we will have two different code paths.